From 23cd1954aba5802493d70742824a6945a80dbe12 Mon Sep 17 00:00:00 2001
From: Meir Levi <mlevi4@marvell.com>
Date: Wed, 26 Jan 2022 10:46:45 +0200
Subject: [PATCH 12/12] octeontx2-af: fix REE boundary graph check

commit f52d9d044a92848bf8b29a81f0400b255c86092e from
git@git.assembla.com:cavium/WindRiver.linux.git

the offset holds memory range (up to 128MB) while the check was
referring to the graph max entries (16M), the fix is multiply
the graph max entries (16M) with the entry size (8B).

Signed-off-by: Meir Levi <mlevi4@marvell.com>
Change-Id: I9e1a86c4cac8f06e5e1f169d76a84232b99094ba
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/69932
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Liron Himi <lironh@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
(cherry picked from commit 09fe0d8a74a3ae55622faf6987d351ba300540ca)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/70020
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_ree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_ree.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_ree.c
index 3c46fa2f06cb..b86e4ae7d258 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_ree.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_ree.c
@@ -457,7 +457,7 @@ int ree_reex_graph_write(struct ree_rsrc *ree,
 
 	while ((*rule_db)->type == REE_ROF_TYPE_7) {
 		offset = ((*rule_db)->addr & 0xFFFFFF) << 3;
-		if (offset > REE_GRAPH_CNT)
+		if (offset > REE_GRAPH_CNT * 8)
 			return REE_AF_ERR_GRAPH_ADDRESS_TOO_BIG;
 		memcpy(ree->graph_ctx->base + offset,
 		       &(*rule_db)->value, sizeof((*rule_db)->value));
-- 
2.34.1

